src/scripts/btma/annotator.py

```diff
@@ -63,81 +65,101 @@ def main():
-                                                           e=exp))
-        tmp = extract_IDs(module)
+                                        --header "{c}" \
+                                        --export {e}'.format(jar=robot_path,
+                                                             m=mod, 
+                                                             c=col, 
+                                                             e=exp))
+        tmp = extract_IDs(csv_path / (module + '.csv'))
         ids = ids.union(set(tmp))
         ids_per_module[i] = tmp
 
     ids = list(ids)
 
-    # Create overview about IRIs and there belonging
+    # Compare new data with database
 
-    print('Finding belongings...')
+    old_ids_per_module = [_,_,_,_]
+    try:
+        for i, module in enumerate(modules):
+            old_ids_per_module[i] = extract_IDs(db_path / (module + '.csv'))
+    except:
+        pass
+        
+    if old_ids_per_module != ids_per_module:
 
-    belonging = ids.copy()
+        # Create overview about IRIs and there belonging
 
-    for (i,id) in enumerate(ids):
-        for (j,module_ids) in enumerate(ids_per_module):
-            if id in module_ids:
-                belonging[i] = modules[j]
-                break
+        print('Finding belongings...')
 
-    # Create the CSV template with the help of the overview
+        belonging = ids.copy()
 
-    print('Creating template...')
+        for (i,id) in enumerate(ids):
+            for (j,module_ids) in enumerate(ids_per_module):
+                if id in module_ids:
+                    belonging[i] = modules[j]
+                    break
+
+        # Create the CSV template with the help of the overview
+
+        print('Creating template...')
         
-    belongs_to_module = list(zip(ids, belonging))
-
-    csv_table = [["ID","belongs to module"],
-                 ["ID","AI OEO:belongs_to_module"]]
-
-    csv_table_per_module = (csv_table, csv_table, csv_table, csv_table)
-
-    for (ids, belonging) in belongs_to_module:
-        if belonging == module[0]:
-            csv_table_per_module[0].append(['OEO:' + ids, ontology_iri + '/' + belonging])
-        elif belonging == module[1]:
-            csv_table_per_module[1].append(['OEO:' + ids, ontology_iri + '/' + belonging])
-        elif belonging == module[2]:
-            csv_table_per_module[2].append(['OEO:' + ids, ontology_iri + '/' + belonging])
-        else:
-            csv_table_per_module[3].append(['OEO:' + ids, ontology_iri + '/' + belonging])
+        belongs_to_module = list(zip(ids, belonging))
+
+        csv_table = [["ID","belongs to module"],
+                     ["ID","AI OEO:belongs_to_module"]]
+
+        csv_table_per_module = (csv_table, csv_table, csv_table, csv_table)
+
+        for (ids, belonging) in belongs_to_module:
+            if belonging == module[0]:
+                csv_table_per_module[0].append(['OEO:' + ids, ontology_iri + '/' + belonging])
```

Comment:
you won't need to add the prefix `OEO:` in these rows if you make the changes suggested above

Reply:
Done